sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FilePartitionReader.scala

```diff
@@ -42,9 +42,7 @@ class FilePartitionReader[T](readers: Iterator[PartitionedFileReader[T]])
           currentReader = getNextReader()
         } catch {
           case e: FileNotFoundException if ignoreMissingFiles =>
-            logWarning(s"Skipped missing file: $currentReader", e)
-            currentReader = null
```

Comment:
How about keeping this line to ensure setting `currentReader` as null

Reply:
+1, it doesn't hurt.